Sources/Basic/FileSystem.swift

```diff
@@ -269,6 +272,20 @@ private class LocalFileSystem: FileSystem {
         let cwdStr = FileManager.default.currentDirectoryPath
         return try? AbsolutePath(validating: cwdStr)
     }
+    
+    var homeDirectory: AbsolutePath? {
+        let homeDirStr: String
+        #if os(macOS)
+            if #available(OSX 10.12, *) {
+                homeDirStr = FileManager.default.homeDirectoryForCurrentUser.path
+            } else {
+                homeDirStr = NSHomeDirectory()
+            }
+        #else
+            homeDirStr = NSHomeDirectory()
+        #endif
+        return try? AbsolutePath(validating: homeDirStr)
```

Comment:
Foundation seems to guarantee a valid path so this doesn't need use the validation version.